R/zigzagHMC.R

```diff
@@ -77,59 +78,59 @@ zigzagHMC <- function(n,
     }
   }
   
-  samples <- array(0, c(n + burnin, ndim))
+  samples <- array(0, c(nSample, ndim))
   
   if (nutsFlg) {
-    if (!is.null(step)) {
-      t <- step
-    } else{
-      t <- 0.1 / sqrt(min(mgcv::slanczos(
+    if (is.null(stepsize)) {
+      stepsize <- 0.1 / sqrt(min(mgcv::slanczos(
         A = prec, k = 1, kl = 1
       )[['values']]))
     }
-    cat("NUTS base step size is", t, "\n")
     engine <- createNutsEngine(
       dimension = ndim,
       lowerBounds = lowerBounds,
       upperBounds = upperBounds,
       flags = 128,
-      seed = rSeed,
-      stepSize = t,
+      seed = seed,
+      stepSize = stepsize,
       mean = mean,
       precision = prec
     )
     
   } else {
-    if (!is.null(step)) {
-      t <- step
-    } else{
-      t <-
+    if (is.null(stepsize)) {
+      stepsize <-
         sqrt(2) / sqrt(min(mgcv::slanczos(
           A = prec, k = 1, kl = 1
         )[['values']], na.rm = T))
```

Comment:
unrelated to this PR, but realized that one of these `slanczos` calls has `na.rm=T` while the other one doesn't, not sure if there was some reason for that.

Reply:
Interesting. Any reason @Zhenyu028?

Reply:
good catch, I don't remember any convergence issue with `slanczos` and `slanczos(...)[['values']]` should be two numbers being the largest and smallest eigenvalues. Though this `na.rm = T` I wrote made me think there may be cases where the largest eigenvalue is null. We can add a check (that slanczos returns two real numbers) here?

Reply:
OK, the convergence issue is less likely for the largest eigenvalue than for the smallest with Lanczos, but I'll watch out for potentially problematic behaviors.